src/charm.py

```diff
@@ -129,29 +131,46 @@ def reconcile(self, event: EventBase) -> None:
         if getattr(event, "departing_unit", None) == self.unit:
             return
 
+        # Maintain the correct status
+        if self.unit.is_leader():
+            if self.state.opensearch_server:
+                clear_status(self.app, MSG_DB_MISSING)
+
+            if self.state.cluster.tls and not self.state.cluster.tls:
+                self.unit.status = MaintenanceStatus(MSG_TLS_CONFIG)
```

Comment:
Let me evalue this (I'm not yet convenienced). More news soon.

Reply:
On the **leader** unit, the flow would be:
1. unit.status = MaintenanceStatus(MSG_TLS_CONFIG)
2. unit.status = MaintenanceStatus(MSG_RECONFIGURING)

In other words, "1." will never be displayed. Did you perhaps mean to set the status to the application instead of the unit?

Reply:
You are right.

Since each reconfiguration actually requires a restart, perhaps it's pointless to keep that status change.
Restrarts anyway set corresponding statuses for each step.

Thus I guess it's best to drop the `MSG_RECONFIGURING` status.

Reply:
Regarding `return`: I've opened this issue https://github.com/canonical/opensearch-dashboards-operator/issues/21 to signify that currently (given the low number of occurances combined with the fast restart speed -- all of which should anyway be hidden behind `traefik`) we choose to restart on the few occasions needed. However may the number grow, this design is to be reviewed.